addons/web/controllers/export.py

```diff
@@ -161,7 +162,7 @@ def insert_leaf(self, group):
             # Update count value and aggregated value.
             node.count += count
 
-        node.data = records.with_env(self._model.env).export_data(self._export_field_names).get('datas', [])
+        node.data = [row for record_id in records.ids for row in self._export_data[record_id]]
```

Comment:
You can actually loop on `_export_data` instead to keep the order of the model, and then you don't need to `search` to reorder `records`.

Reply:
Thanks for the change. The change behavior (keeping the order of ids selected when groupby) makes perfectly sense to me, it is now consistent with the no-groupby export. :+1: